packages/browser/src/core/user/index.ts

```diff
@@ -183,18 +200,92 @@ export interface CookieOptions {
   sameSite?: string
 }
 
```

Comment:
This is neat. Is there a way we can make this fully generic -- like I did with the [event emitter](https://github.com/segmentio/analytics-next/blob/3358219e1757384f3be66109971a1c9e5f384dda/packages/core/src/emitter/index.ts#L18)?


```ts
export class UniversalStorage<Data extends Record<string, unknown>> {
..

public get<K extends keyof Data>(
    key: K,
    storeTypes?: StoreType[]
  ): Data[K] | null {
    let val = null
    for (const store of this.getStores(storeTypes)) {
      val = store.get<Data[K]>(key as string) // store can be made generic too
      if (val) {
        return val
      }
    }
    return null
  }
```

would lead to something like:

![image](https://user-images.githubusercontent.com/5115498/198116858-89329bdb-ee53-42de-93ae-009b51289c20.png)
![image](https://user-images.githubusercontent.com/5115498/198116875-128323dd-1198-47eb-a4bc-44bc136323d4.png)

Reply:
Great suggestion. Added. Could you please if there is any TS smells in my changes or anything can be improved?

Reply:
I don't see any of the changes yet -- did you push?

Reply:
Yes, here is the commit: https://github.com/segmentio/analytics-next/pull/638/commits/c17e5fc572f8071dfa31572e4be6145db8e670f4

Reply:
I see the build error --  you could force people to always explicitly pass data or, if we don't care, we might want to add add a default generic parameter like so:
`UniversalStorage<Data extends Record<string, unknown> = Record<string, unknown>>`

Reply:
> I know one person that really thinks default generic parameters can be sometimes "harmful"

What? But they're so handy!